roofit/roofitcore/inc/RooFit/Detail/NormalizationHelpers.h

```diff
@@ -70,8 +70,7 @@ template <class T>
 std::unique_ptr<T> compileForNormSet(T const &arg, RooArgSet const &normSet)
 {
    RooFit::Detail::CompileContext ctx{normSet};
-   std::unique_ptr<RooAbsArg> head = arg.compileForNormSet(normSet, ctx);
-   return std::unique_ptr<T>{static_cast<T *>(head.release())};
+   return std::unique_ptr<T>{static_cast<T *>(arg.compileForNormSet(normSet, ctx).release())};
```

Comment:
Could you please comment about the need of this change?

Reply:
Ok, let me explain the need of this change.

In the code without my patch, `std::unque_ptr<RooAbsArg> head` is constructed in the function, and it is needed to be destructed at the end of the function. However, `RooAbsArg` was still not defined, so it could not be destructed. In my test (gcc 13, with `-std=c++23`), this will cause a compile error.

So, generally speaking, we should move the function body to c++ source file. However, this is a function template, simply move the definition into source file will lead to failure in [template implicit instantiation](https://en.cppreference.com/w/cpp/language/function_template#Implicit_instantiation). Thus, I removed the definition of local variable `head` to avoid calling to its destrutor.

Of course, the return value of `arg.compileForNormSet(normSet, ctx)` is typed `std::unqie_ptr` and should also be destructed in this function, but this will not cause a compile error in my test. To my understand, this is because compiler did not know the type of the return value of `arg.compileForNormSet` before `T` is specified (i.e., before the function template is called and instantiated), so this statement will only be compiled when `T` is specified (i.e. when the function is actually called, where the template is implicitely instantiated). I am not sure this is a compiler specified behavior or in C++ standard.